src/core/jit/model.rs

```diff
@@ -703,4 +714,29 @@ mod test {
         let async_directive: ConstDirective = (&custom_directive).into();
         insta::assert_debug_snapshot!(async_directive);
     }
+    #[test]
+    fn test_operation_plan_dedupe() {
+        let config = include_config!("../../../tailcall-fixtures/fixtures/configs/jsonplaceholder.graphql").unwrap();
+        let module = ConfigModule::from(config);
+        let bp = Blueprint::try_from(&module).unwrap();
+
+        let request = Request::new(r#"{ posts { id } }"#);
+        let jit_request = jit::Request::from(request);
+        let plan = jit_request.create_plan(&bp).unwrap();
+
+        assert!(plan.dedupe);
+    }
+
+    #[test]
+    fn test_operation_plan_dedupe_false() {
+        let config = include_config!("../../../tailcall-fixtures/fixtures/configs/jsonplaceholder.graphql").unwrap();
+        let module = ConfigModule::from(config);
+        let bp = Blueprint::try_from(&module).unwrap();
+
+        let request = Request::new(r#"{ users { id } }"#);
+        let jit_request = jit::Request::from(request);
+        let plan = jit_request.create_plan(&bp).unwrap();
+
+        assert!(!plan.dedupe);
+    }
```

Comment:
add another test where there are nested calls and some are deduped and some are not.

Reply:
it's checked on flattened form of field